src/general/DayNightCycle.cs

```diff
@@ -0,0 +1,91 @@
+﻿using System;
+using Newtonsoft.Json;
+
+[JsonObject(IsReference = true)]
+[UseThriveSerializer]
+public class DayNightCycle
+{
+    [JsonProperty]
+    public DayNightConfiguration LightCycleConfig;
+
+    [JsonProperty]
+    public bool IsEnabled;
+
+    /// <summary>
+    ///   The multiplier used for calculating DayLightPercentage.
+    /// </summary>
+    /// <remarks>
+    ///   This exists as it only needs to be calculated once and
+    ///   the calculation for it is confusing.
+    /// </remarks>
+    [JsonIgnore]
+    private float daytimeMultiplier;
+
+    public DayNightCycle(bool isEnabled)
+    {
+        IsEnabled = isEnabled;
+
+        LightCycleConfig = SimulationParameters.Instance.GetDayNightCycleConfiguration();
+        Time = LightCycleConfig.HoursPerDay / 2;
+
+        float halfPercentage = LightCycleConfig.DaytimePercentage / 2;
+
+        // This converts the percentage in DaytimePercentage to the power of two needed for DayLightPercentage
+        daytimeMultiplier = (float)Math.Pow(2, 1 / halfPercentage);
+
+        if (IsEnabled)
+        {
+            AverageSunlight = EvaluateAverageSunlight(0.5f + halfPercentage)
+                - EvaluateAverageSunlight(0.5f - halfPercentage);
```

Comment:
These lines confuse me a lot. What I read here is that the average sunlight is a difference between average sunlight (`EvaluateAverageSunlight`), which seems off. But I think that `EvaluateAverageSunlight` actually computes the *accumulated* sunlight throughout the day (that is, the antiderivative of the sunlight value function).

Also, the fact that you (rightfully) dropped the nighttime parts for simpler computations should probably warrant a small comment stating that.

Reply:
I struggled a lot with naming that function. Suggestions welcome.

Reply:
`CumulativeSunlightValue`? Or, perhaps more visually `TotalSunlightSinceMidnight`?